src/lib/misc.c

```diff
@@ -793,7 +794,7 @@ pgp_mem_readfile(pgp_memory_t *mem, const char *f)
         mem->mmapped = 1;
     }
     (void) fclose(fp);
-    return (mem->allocated == mem->length);
+    return (mem->allocated == mem->length) ? RNP_OK : RNP_FAIL;
```

Comment:
Logically this function returns error code (0 for failure, 1 for success, probably later other int for more specific error code), while (mem->allocated == mem->length) logically is bool, not int.